compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -274,6 +276,21 @@ BLangStatementExpression desugar(BLangQueryExpr queryExpr, SymbolEnv env,
                             queryExpr.getBType()));
             streamStmtExpr.setBType(tableRef.getBType());
             onConflictExpr = null;
+        } else if (queryExpr.isMap) {
+            onConflictExpr = (onConflictExpr == null)
+                    ? ASTBuilderUtil.createLiteral(pos, symTable.nilType, Names.NIL_VALUE)
+                    : onConflictExpr;
+            BMapType mapType = (BMapType) ((BUnionType) queryExpr.getBType()).getMemberTypes()
```

Comment:
Here since we get the first possible match; can there be complications? Shall we add a test with two un-assignable map types

Reply:
We cannot add tests for this scenario due to this issue https://github.com/ballerina-platform/ballerina-lang/issues/36476. 
Probably we have to check this after fixing above issue.